File: api/web/models/job.ex
defmodule Jobs.Job do
  use Jobs.Web, :model

  schema "jobs" do
    field :title, :string
    field :short_description, :string
    field :description, :string
    field :starts_at, Ecto.DateTime
    field :ends_at, Ecto.DateTime
    field :slug, :string
    belongs_to :user, Jobs.User
    belongs_to :organization, Jobs.Organization

    timestamps
  end

  @required_fields ~w(title short_description description starts_at ends_at slug)
  @optional_fields ~w()

  @doc """
  Creates a changeset based on the `model` and `params`.

  If no params are provided, an invalid changeset is returned
  with no validation performed.
  """
  def changeset(model, params \\ :empty) do
    model
    |> cast(params, @required_fields, @optional_fields)
  end
end

File: client/app/models/job.js
import Model from 'ember-data/model';
import attr from 'ember-data/attr';
import DS from 'ember-data';

export default Model.extend({
  title: attr('string'),
  shortDescription: attr('text'),
  description: attr('text'),
  startsAt: attr('datetime'),
  endsAt: attr('datetime'),
  slug: attr('string'),
  user: DS.belongsTo('user', { async: true }),
  organization: DS.belongsTo('organization', { async: true })
});
